src/preact/component/types.d.ts

```diff
@@ -0,0 +1,59 @@
+import * as Preact from 'preact';
+import {ReadyState_Enum} from '#core/constants/ready-state';
+import {Ref, VNode, FunctionComponent, ComponentChildren} from 'preact';
+
+// TODO: figure out the proper type for all of the 'as' types.
+
+export interface WrapperComponentProps {
+  style: Preact.JSX.CSSProperties;
+  class: string;
+  as: FunctionComponent | any;
+  wrapperClassName?: string | null;
+  wrapperStyle?: Object | null;
+  children?: ComponentChildren;
+}
+
+/**
+ * See https://developer.mozilla.org/en-US/docs/Web/CSS/contain
+ */
+export interface ContainerWrapperComponentProps {
+  as?: FunctionComponent | any;
+  contentAs?: string | FunctionComponent;
+  contentProps?: any;
+  style?: Preact.JSX.CSSProperties;
+  size?: boolean;
+  layout?: boolean;
+  paint?: boolean;
+  wrapperClassName?: string | null;
+  wrapperStyle?: Object | null ;
+  contentRef?: null | Ref<any>;
+  contentClassName?: string | null ;
+  contentStyle?: Object | null;
+  children?: ComponentChildren | null;
+  class?: string | null;
+}
+
+export type RendererFunction = (obj: JsonObject) => VNode | Promise<VNode>;
+
+export interface IframeEmbedProps {
+  allow: string | undefined;
+  allowFullScreen: boolean | undefined;
+  loading: string;
+  messageHandler: (ref: Ref<HTMLIFrameElement>) => void;
+  matchesMessagingOrigin: (a:any) => boolean;
+  name: string | undefined;
+  onReadyState: (str: string) => void | undefined;
+  ready: boolean | undefined;
+  sandbox: string | undefined;
+  src: string | undefined;
+  title: string | undefined;
```

Comment:
Is it really that `title` always set and is sometimes `undefined`, or should it be `title?: string` for most of these?

Reply:
I think almost everywhere with `|undefined` it was probably meant to be an optional arg